src/Adapter/PhpSerialize.php

```diff
@@ -64,8 +64,8 @@ public function serialize($value)
      */
     public function unserialize($serialized)
     {
-        if (!is_string($serialized) || !preg_match('/^((s|i|d|b|a|O|C):|N;)/', $serialized)) {
```

Comment:
I do note understand why you're removing this check.

The regular expression there ensures that we don't run an `unserialize()` call unnecessarily; if the string is clearly invalid, there's no reason to make the attempt.

We can still throw the exception; I just think we should keep the `preg_match` *too*.